ts/collection.ts

```diff
@@ -196,11 +202,50 @@ export class Collection<T extends any> {
     return parse(await this._find(filter, { findOne: true }))[0] ?? null;
   }
 
-  public async find(
-    filter?: FilterType<T>,
-    options?: FindOptions,
-  ): Promise<Array<T & WithID>> {
-    return parse(await this._find(filter, { findOne: false, ...options }));
+  public find(filter?: FilterType<T>, options?: FindOptions): Cursor {
+    return new class extends Cursor {
+      private maxQueryLimit: number = -1;
+      private skipDocCount: number = -1;
+      private superClass: any;
+
+      constructor(superClass: any) {
+        super();
+        this.superClass = superClass;
+      }
+
+      private async _get(): Promise<(T & WithID) | null> {
+        return parse(
+          await this.superClass._find(filter, {
+            findOne: false,
+            limit: this.maxQueryLimit < 0 ? undefined : this.maxQueryLimit,
+            skip: this.skipDocCount < 0 ? undefined : this.skipDocCount,
+            ...options,
+          }),
+        );
+      }
+
+      public then(callback: any): Promise<(T & WithID) | null> {
+        return this._get().then(callback);
+      }
+
+      public catch(callback: any): Promise<(T & WithID) | null> {
+        return this._get().catch(callback);
+      }
+
+      public finally(callback: any): Promise<(T & WithID) | null> {
+        return this._get().finally(callback);
+      }
```

Comment:
If you execute 
```ts
users.find().skip(1).then(console.log).catch(console.error);
```

`then` and `catch` each executed a query, which is not expected which is not expected